src/python/gudhi/simplex_tree_multi.pyx

```diff
@@ -0,0 +1,970 @@
+
+		:param other: If `other` is `None` (default value), an empty `SimplexTree` is created.
+			If `other` is a `SimplexTree`, the `SimplexTree` is constructed from a deep copy of `other`.
+		:type other: SimplexTree (Optional)
+		:returns: An empty or a copy simplex tree.
+		:rtype: SimplexTree
+
+		:raises TypeError: In case `other` is neither `None`, nor a `SimplexTree`.
+		:note: If the `SimplexTree` is a copy, the persistence information is not copied. If you need it in the clone,
+			you have to call :func:`compute_persistence` on it even if you had already computed it in the original.
+		"""
+		
+	# The real cython constructor
+	def __cinit__(self, other = None, num_parameters:int=2): #TODO doc
+		if not other is None:
+			if isinstance(other, SimplexTreeMulti):
+				self.thisptr = _get_copy_intptr(other)
+			elif isinstance(other, SimplexTree): # Constructs a SimplexTreeMulti from a SimplexTree
+				self.thisptr = <intptr_t>(new Simplex_tree_multi_interface())
+				multify(other.thisptr, self.thisptr, num_parameters)
+			else:
+				raise TypeError("`other` argument requires to be of type `SimplexTree`, or `None`.")
+		else:
+			self.thisptr = <intptr_t>(new Simplex_tree_multi_interface())
+		self.get_ptr().set_number_of_parameters(num_parameters)
+	
+	# TODO : set number of parameters outside the constructor ?
+
+	def __dealloc__(self):
+		cdef Simplex_tree_multi_interface* ptr = self.get_ptr()
+		if ptr != NULL:
+			del ptr
+		# if self.pcohptr != NULL:
+		#     del self.pcohptr
+
+	def __is_defined(self):
+		"""Returns true if SimplexTree pointer is not NULL.
+			"""
+		return self.get_ptr() != NULL
+
+	# def __is_persistence_defined(self):
+	#     """Returns true if Persistence pointer is not NULL.
+	#      """
+	#     return self.pcohptr != NULL
+
+	def copy(self)->SimplexTreeMulti:
+		"""
+		:returns: A simplex tree that is a deep copy of itself.
+		:rtype: SimplexTree
+
+		:note: The persistence information is not copied. If you need it in the clone, you have to call
+			:func:`compute_persistence` on it even if you had already computed it in the original.
+		"""
+		stree = SimplexTreeMulti(num_parameters=self.num_parameters)
+		stree.thisptr = _get_copy_intptr(self)
+		return stree
+
+	def __deepcopy__(self):
+		return self.copy()
+
+	def filtration(self, simplex)->filtration_type:
+		"""This function returns the filtration value for a given N-simplex in
+		this simplicial complex, or +infinity if it is not in the complex.
+
+		:param simplex: The N-simplex, represented by a list of vertex.
+		:type simplex: list of int
+		:returns:  The simplicial complex filtration value.
+		:rtype:  float
+		"""
+		# filtration_vector = self.get_ptr().simplex_filtration(simplex)
+		# filtrations = np.array_split(filtration_vector, len(filtration_vector) // self.num_parameters)
+		# return filtrations.squeeze() # If 1 critical returns only 1 filtration
+		return self.get_ptr().simplex_filtration(simplex) # Same output type, better
```

Comment:
I like `reshape((-1, num_parameters))`, it makes it clear that there are multiple critical points.
`squeeze` has the problem that you end up having to test the size of the output for every call. If the whole filtration is 1-critical, it may be convenient to avoid the extra dimension, but I don't think it is worth the resulting inconsistency.